cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DocumentStore.java

```diff
@@ -0,0 +1,138 @@
+package com.cloudant.sync.datastore;
+
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.datastore.encryption.NullKeyProvider;
+import com.cloudant.sync.event.EventBus;
+import com.cloudant.sync.event.Subscribe;
+import com.cloudant.sync.notifications.DatabaseClosed;
+import com.cloudant.sync.notifications.DatabaseCreated;
+import com.cloudant.sync.notifications.DatabaseDeleted;
+import com.cloudant.sync.notifications.DatabaseOpened;
+import com.cloudant.sync.query.IndexManager;
+import com.cloudant.sync.query.IndexManagerImpl;
+import com.cloudant.sync.util.Misc;
+import com.sun.javafx.scene.shape.PathUtils;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Created by tomblench on 27/09/2016.
+ */
```

Comment:
@emlaver done in 0f71e9d

Some methods still need documenting, we'll cover these in a documentation pass on a later PR.